### PR TITLE
Add option to filter PRs based on ready state

### PR DESCRIPTION
**Pull request by pjohnmeyer**
Implements #176.

b83873d goes against the style of the rest of the tests a little, so if you want me to revert it out and make the tests more linear I can. I just wanted to do a _little_ refactoring of the work I did before I put this up.

### CODE REVIEW

File: src/autoupdater.ts [hunk 1]
```diff
@@ -331,6 +331,21 @@ export class AutoUpdater {
...
+        ghCore.info('Pull request is not draft, skipping update.');
+        return false;
+      }
+
+      if (readyStateFilter === 'ready_for_review' && pull.draft) {
+        ghCore.info('Pull request is draft, skipping update.');
+        return false;
+      }
+    }
+
```
Comment (pjohnmeyer) [inline src/autoupdater.ts:348]: It occurs to me that this check needs to move up before any `return true` code paths to work correctly.  I’ll make that change in the coming days (or you can @chinthakagodawita if you’re so inclined).

Comment (pjohnmeyer) [reply src/autoupdater.ts]: Done.

File: src/autoupdater.ts [hunk 2]
```diff
@@ -265,6 +265,21 @@ export class AutoUpdater {
       }
     }
 
+    const readyStateFilter = this.config.pullRequestReadyState();
+    if (readyStateFilter !== 'all') {
+      ghCore.info('Checking PR ready state');
+
+      if (readyStateFilter === 'draft' && !pull.draft) {
+        ghCore.info('Pull request is not draft, skipping update.');
```
Comment (chinthakagodawita) [inline src/autoupdater.ts:273]: Just a minor recommendation to make it slightly more verbose:
```suggestion
        ghCore.info('PR_READY_STATE=draft and pull request is not draft, skipping update.');
```

File: src/autoupdater.ts [hunk 3]
```diff
@@ -265,6 +265,21 @@ export class AutoUpdater {
...
+        return false;
+      }
+
+      if (readyStateFilter === 'ready_for_review' && pull.draft) {
+        ghCore.info('Pull request is draft, skipping update.');
```
Comment (chinthakagodawita) [inline src/autoupdater.ts:278]: Same deal here:
```suggestion
        ghCore.info('PR_READY_STATE=ready_for_review and pull request is draft, skipping update.');
```

**Review by chinthakagodawita — COMMENTED** [review]
Nice and simple, great work! Just have a very minor comment - let me know what you think

**Comment by chinthakagodawita** [comment]
I've also tested this manually and it does exactly what it says on the tin 🎉

**Comment by pjohnmeyer** [comment]
@chinthakagodawita incorporated your suggestions, thanks!

**Review by chinthakagodawita — APPROVED** [review]
LGTM!

**Comment by chinthakagodawita** [comment]
Released in [v1.5.0](https://github.com/chinthakagodawita/autoupdate/releases/tag/v1.5.0), thanks again! 🙌

**Pull request merged**